### PR TITLE
doc: note that our ChaCha20 isn't standard compliant.

### LINKED ISSUE #21095: EVP_chacha20 does not match RFC 7539 or documentation in nonce/counter split
RFC 7539 specified ChaCha20 with a 96-bit nonce and a 32-bit counter. Only 32 bits of the input is incremented across each block.
https://www.rfc-editor.org/rfc/rfc7539.html#section-2.4

The original paper didn't specify a split but, prior to the RFC, 64-bit nonce with 64-bit counter was also common. A 96/32 and 64/64 split are very similar primitives, but not quite the same. OpenSSL's documentation says it implements the RFC formulation:

> The ChaCha20 stream cipher. The key length is 256 bits, the IV is 128 bits long. The first 32 bits consists of a counter in little-endian order followed by a 96 bit nonce.

https://www.openssl.org/docs/manmaster/man3/EVP_chacha20.html

However, the implementation doesn't match this and uses a 64-bit counter.
https://github.com/openssl/openssl/blob/b134300a342476398c11c19af602d7b2aa6b7f8a/crypto/evp/e_chacha20_poly1305.c#L112-L113

See also https://github.com/google/wycheproof/issues/90

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #21095

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/EVP_chacha20.pod [hunk 1]
```diff
@@ -22,7 +22,7 @@ The ChaCha20 stream cipher for EVP.
 =item EVP_chacha20()
 
 The ChaCha20 stream cipher. The key length is 256 bits, the IV is 128 bits long.
-The first 32 bits consists of a counter in little-endian order followed by a 96
+The first 64 bits consists of a counter in little-endian order followed by a 64
 bit nonce. For example a nonce of:
 
 000000000000000000000002
```
Comment (t8m) [inline doc/man3/EVP_chacha20.pod:28]: You need to change this to 64 bits value as well and the value below too.

**Comment by levitte** [comment]
Kicking the CIs

File: doc/man3/EVP_chacha20.pod (hunk 1 above)
Comment (paulidale) [reply doc/man3/EVP_chacha20.pod]: Good catch.
Fixed.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Ping @openssl/committers for 2nd review.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**